src/app/common/utils.ts

```diff
@@ -312,6 +312,11 @@ export function whenStxChainId(chainId: ChainID) {
   return <T>(chainIdMap: WhenChainIdMap<T>): T => chainIdMap[chainId];
 }
 
+type NetworkMap<T> = Record<NetworkModes, T>;
+export function whenNetwork(mode: NetworkModes) {
```

Comment:
Maybe the `when...` helper functions should start being pulled out into their own file?

Reply:
I've been thinking about the common folder in general. Think we could organise better.

Perhaps at a higher level organising between:
- domain-specific logic (anything stx/btc related)
- language helpers (`isString` etc)
- ui helpers (`truncateMiddle`)

Reply:
Coming back here, I like this idea. I agree the `common` folder needs more structure. I'll make an issue for this if there isn't one already.